File: Hackerrank/day10/main.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())
    binario = bin(n)[2:]

    grupos_de_unos = binario.split('0')

    max_unos = max(len(grupo) for grupo in grupos_de_unos)

    print(max_unos)

File: Hackerrank/day10/main.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    const n = parseInt(readLine().trim(), 10);
    const binario = n.toString(2);

    const grupos = binario.split('0');

    const maxUnos = Math.max(...grupos.map(g => g.length));

    console.log(maxUnos);
}
